Extensions/TweenBehavior/JsExtension.js

```diff
@@ -0,0 +1,397 @@
+ * and search for any errors.
+ *
+ * More information on https://github.com/4ian/GDevelop/blob/master/newIDE/README-extensions.md
+ */
+module.exports = {
+  createExtension: function (t, gd) {
+    const extension = new gd.PlatformExtension();
+    extension
+      .setExtensionInformation(
+        "Tween",
+        t("Tween animation"),
+        t("Animate object properties via tweening."),
+        "Matthias Meike",
+        "Open source (MIT License)"
+      )
+      .setExtensionHelpPath("/all-features/tween");
+
+    var tweenBehavior = new gd.BehaviorJsImplementation();
+
+    tweenBehavior.updateProperty = function (
+      behaviorContent,
+      propertyName,
+      newValue
+    ) {
+      if (propertyName === "My first property") {
+        behaviorContent.property1 = newValue;
+        return true;
+      }
+      if (propertyName === "My other property") {
+        behaviorContent.property2 = newValue === "1";
+        return true;
+      }
+
+      return false;
+    };
+
+    tweenBehavior.getProperties = function (behaviorContent) {
+      var behaviorProperties = new gd.MapStringPropertyDescriptor();
+
+      behaviorProperties.set(
+        "My first property",
+        new gd.PropertyDescriptor(behaviorContent.property1)
+      );
+      behaviorProperties.set(
+        "My other property",
+        new gd.PropertyDescriptor(
+          behaviorContent.property2 ? "true" : "false"
+        ).setType("Boolean")
+      );
+
+      return behaviorProperties;
+    };
+
+    tweenBehavior.setRawJSONContent(
+      JSON.stringify({
+        property1: "Initial value 1",
+        property2: true
+      })
+    );
+
+   const behavior = extension
+      .addBehavior(
+        "TweenBehavior",
+        t("Tween animation"),
+        "TweenBehavior",
+        t("Tween animation"),
+        "",
+        "CppPlatform/Extensions/topdownmovementicon.png",
+        "TweenBehavior",
+        tweenBehavior,
+        new gd.BehaviorsSharedData()
+      )
+      .setIncludeFile("Extensions/TweenBehavior/tweenruntimebehavior.js")
+      .addIncludeFile("Extensions/TweenBehavior/shifty.js");
+    //.addIncludeFile("Extensions/TweenBehavior/LICENSE-MIT.txt");
+
+    // Behavior related
+    behavior.addAction(
+        "AddObjectVariableTween",
+        t("Add object variable tween"),
+        t("Add a tween animation for an object variable."),
+        t(
+          "Tween the variable _PARAM3_ of _PARAM0_ from _PARAM4_ to _PARAM5_ with easing _PARAM6_ over _PARAM7_ms as _PARAM2_"
+        ),
+        t("Tween animation/Object variables"),
+        "JsPlatform/Extensions/take_screenshot24.png",
+        "JsPlatform/Extensions/take_screenshot32.png"
+      )
+      .addParameter("object", t("Object"), "", false)
+      .addParameter("behavior", t("Behavior"), "TweenBehavior", false)
+      .addParameter("string", t("Tween Identifier"), "", false) // TODO: Change into stringlist
+      .addParameter("objectvar", t("Object variable"), "", false)
```

Comment:
So I've checked and the code generator expect the object parameter to be *just* before the "objectvar". Meaning that this won't work here, and can't work because the second argument is always "behavior" by convention.

So we're blocked to get this working for now! To fix this, I'll have to update the code generator (in libGD.js) to read previous parameters. I'll probably do an update of libGD.js that also include the "stringlist" parameter type.

Reply:
While you are at it, could you also add a `getColor` function to the sprite object? (There is currently only a [setColor ](https://github.com/4ian/GDevelop/blob/master/GDJS/Runtime/spriteruntimeobject.js#L726)function)
It would be possible to tween object colors then. Would come in handy for get hurt animations and stuff. We could also show the current sprite color in the IDE's setColor action editor.